src/main/java/teammates/logic/api/LogsProcessor.java

```diff
@@ -84,8 +84,8 @@ public int getNumberOfLogsForEvent(Instant startTime, Instant endTime, LogEvent
             QueryLogsResults logFetchResults = queryLogs(logsParams);
             List<GeneralLogEntry> logs = logFetchResults.getLogEntries();
             total += logs.size();
-            if (logFetchResults.getHasNextPage()) {
-                logStartTime = logs.get(logs.size() - 1).getTimestamp();
+            if (logFetchResults.getHasNextPage() && !logs.isEmpty()) {
+                logStartTime = logs.getLast().getTimestamp();
```

Comment:
getLast() does not exist for Lists in JDK 11

it was added in JDK 21 https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/util/List.html#getLast()

Reply:
switched back to previous method